data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/cnec/CsaProfileFlowCnecCreator.java

```diff
@@ -0,0 +1,390 @@
+            return false;
+        }
+    }
+
+    private Double getCurrentLimit(Branch<?> branch, Branch.Side side) {
+
+        if (branch.getCurrentLimits(side).isPresent()) {
+            return branch.getCurrentLimits(side).orElseThrow().getPermanentLimit();
+        }
+
+        if (side == Branch.Side.ONE && branch.getCurrentLimits(Branch.Side.TWO).isPresent()) {
+            return branch.getCurrentLimits(Branch.Side.TWO).orElseThrow().getPermanentLimit() * branch.getTerminal1().getVoltageLevel().getNominalV() / branch.getTerminal2().getVoltageLevel().getNominalV();
+        }
+
+        if (side == Branch.Side.TWO && branch.getCurrentLimits(Branch.Side.ONE).isPresent()) {
+            return branch.getCurrentLimits(Branch.Side.ONE).orElseThrow().getPermanentLimit() * branch.getTerminal2().getVoltageLevel().getNominalV() / branch.getTerminal1().getVoltageLevel().getNominalV();
+        }
+
+        return null;
+    }
+
+    private Identifiable<?> getNetworkElementInNetwork(String networkElementId) {
+        Identifiable<?> networkElement = network.getIdentifiable(networkElementId);
+        if (networkElement == null) {
+            CgmesBranchHelper cgmesBranchHelper = new CgmesBranchHelper(networkElementId, network);
+            if (cgmesBranchHelper.isValid()) {
+                networkElement = cgmesBranchHelper.getBranch();
+            }
+        }
+
+        if (networkElement instanceof DanglingLine) {
+            Optional<TieLine> optionalTieLine = ((DanglingLine) networkElement).getTieLine();
+            if (optionalTieLine.isPresent()) {
+                networkElement = optionalTieLine.get();
+            }
+        }
+        return networkElement;
+    }
+
+    private boolean addInstant(String assessedElementId, FlowCnecAdder flowCnecAdder, PropertyBag currentLimit, Boolean isCombinableWithContingency) {
+        String kind = currentLimit.get(CsaProfileConstants.REQUEST_CURRENT_LIMIT_OPERATIONAL_LIMIT_KIND);
+        Instant instant = null;
+
+        if (CsaProfileConstants.OPERATIONAL_LIMIT_TYPE_TATL.equals(kind)) {
+            String acceptableDurationStr = currentLimit.get(CsaProfileConstants.REQUEST_CURRENT_LIMIT_OPERATIONAL_LIMIT_ACCEPTABLE_DURATION);
+            Double acceptableDuration = Double.valueOf(acceptableDurationStr);
+            if (acceptableDuration == null || acceptableDuration < 0) {
+                csaProfileFlowCnecCreationContexts.add(CsaProfileFlowCnecCreationContext.notImported(assessedElementId, ImportStatus.INCONSISTENCY_IN_DATA, "OperationalLimitType.acceptableDuration is incorrect : " + acceptableDurationStr));
+                return false;
+            } else if (acceptableDuration == 0) {
+                instant = isCombinableWithContingency ? Instant.PREVENTIVE : Instant.CURATIVE;
```

Comment:
i think you should check here isBaseCase: to create a preventive CNEC you need preventive thresholds and isBaseCase flag

Reply:
ok done